src/main/webapp/query-control.js

```diff
@@ -216,6 +223,68 @@ async function createStateDropdownList() {
   });
 }
 
+// Set up an autocomplete dropdown, attached to the main location input
+// and states dropdown, to suggest places to users as they type
+function setupAutocompleteLocation() {
+  const autocompleteInput = document.getElementById('location-list');
+  const autocompleteResults = document.getElementById('location');
+  const service = new google.maps.places.AutocompleteService();
+  const defaultLocationOptions = autocompleteResults.innerHTML; // all states
+
+  function displaySuggestions(predictions, status) {
+    if (predictions === null) { // No predictions generated for this location
+      return;
+    }
+    // Each time we display suggestions, show only the top 5, and then the
+    // normal state dropdown afterwards
+    predictions.splice(5);
+    const resultsHtml = [];
+    predictions.forEach(function(prediction) {
+      resultsHtml.push(
+          '<option class="autocomplete-item" value="' +
+          prediction.description +
+          '"></option>');
+    });
+    autocompleteResults.innerHTML =
+        resultsHtml.join('') + defaultLocationOptions;
+  };
+
+  // When the input box changes (due to typing), get what has been typed and
+  // send it to the autocomplete service, which then calls displaySuggestions
+  // on the predictions it generated
+  autocompleteInput.addEventListener('input', debounce(function() {
+    const value = this.value;
+    value.replace('"', '\\"').replace(/^\s+|\s+$/g, ''); // Trim whitespace
+    service.getPlacePredictions({
+      'input': value,
+      'types': ['(regions)'], // no businesses, just cities counties etc.
+      'componentRestrictions': {'country': 'us'}, // USA only
+    }, displaySuggestions);
+  }, 175)); // wait this many ms to send request so they don't overload
```

Comment:
I didn't see earlier versions but I think 175 is good, I'm not seeing any shaking and I can both type fast and see results if I wait.